src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -100,4 +100,33 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             return response;
         }
     }
+    
+    // Custom web hooks for extensions. 
+    public class HookController : ApiController
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+
+        public HookController(WebScriptHostManager scriptHostManager)
+        {
+            _scriptHostManager = scriptHostManager;
+        }
+                
+        [Route("hook/{name}")]
+        [HttpGet]
+        [HttpPost]
+        [AllowAnonymous]
```

Comment:
The extension / receiver is responsible for auth.   The extension is the one handing out access to this URL, so it can do things like 1) append a key and check it, 2) manually check the request.